pkg/controller/redirect/helpers.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 Google LLC
```

Comment:
These are moved into their own file so we can test them. The integration tests need an `_test` package name to prevent the import cycle.